### PR TITLE
feat(python-binding)

### PR DESCRIPTION
**Pull request by Senhaji-Rhazi-Hamza**
add collection interface with methods : 

+ insert_one 
+ find_one 

little demo : https://sharing.clickup.com/clip/p/t2195826/74854463-3ade-40e4-a369-27c71be4e496/screen-recording-2024-09-29-00%3A23.webm

### CODE REVIEW

File: py-polodb/Cargo.toml [hunk 1]
```diff
@@ -19,4 +19,5 @@ crate-type = ["cdylib", "rlib"]
 [dependencies]
 # polodb_core = "5.1.0"
 polodb_core = { path = "../src/polodb_core", default-features = false  }
-pyo3 = { version = "0.22.2", features = ["extension-module"] }
+pyo3 = { version = "0.22.2", features = ["extension-module", "auto-initialize"] }
+bson = "2.0"
```
Comment (vincentdchan) [inline py-polodb/Cargo.toml:23]: Do we really need this? `polodb_core` has an export bson module: `bson_core::bson`

File: py-polodb/src/py_database.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+use std::borrow::Borrow;
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+
+use crate::doc_bson_to_py_translator::{bson_to_py_obj,convert_py_obj_to_document,document_to_pydict};
+
+
+
```
Comment (vincentdchan) [inline py-polodb/src/py_database.rs:15]: It seems there are too many spaces here. Can we leave only 1-2 empty lines here?

File: py-polodb/Cargo.toml (hunk 1 above)
Comment (Senhaji-Rhazi-Hamza) [reply py-polodb/Cargo.toml]: removed

File: py-polodb/src/py_database.rs (hunk 2 above)
Comment (Senhaji-Rhazi-Hamza) [reply py-polodb/src/py_database.rs]: sûre, reformatted

**Pull request merged**